src/Microsoft.IdentityModel.Tokens/Base64UrlEncoding.cs

```diff
@@ -162,106 +141,20 @@ public static T Decode<T, TX, TY, TZ>(
         {
             _ = action ?? throw LogHelper.LogArgumentNullException(nameof(action));
 
-            ReadOnlySpan<char> inputSpan = input.AsSpan();
-            int outputsize = ValidateAndGetOutputSize(inputSpan, offset, length);
-            byte[] output = ArrayPool<byte>.Shared.Rent(outputsize);
+            ReadOnlySpan<char> slice = Validate(input.AsSpan(), offset, length);
+            int outputsize = Base64Url.GetMaxDecodedLength(length);
+            byte[] destination = ArrayPool<byte>.Shared.Rent(outputsize);
+
             try
             {
-                Decode(inputSpan, offset, length, output);
-                return action(output, outputsize, argx, argy, argz);
+                OperationStatus status = Base64Url.DecodeFromChars(slice, destination, out _, out int bytesWritten);
+                if (status != OperationStatus.Done)
+                    throw LogHelper.LogExceptionMessage(new FormatException(LogHelper.FormatInvariant(LogMessages.IDX10400, slice.ToString())));
+                return action(destination, bytesWritten, argx, argy, argz);
             }
             finally
             {
-                ArrayPool<byte>.Shared.Return(output);
-            }
-        }
-
-        /// <summary>
-        /// Decodes a Base64Url encoded substring of a string into a byte array.
-        /// </summary>
-        /// <param name="input">The string represented as a span to decode.</param>
-        /// <param name="offset">The index of the character in <paramref name="input"/> to start decoding from.</param>
-        /// <param name="length">The number of characters beginning from <paramref name="offset"/> to decode.</param>
-        /// <param name="output">The byte array to place the decoded results into.</param>
-        /// <remarks>
-        /// Changes from Base64UrlEncoder implementation:
-        /// 1. Padding is optional.
-        /// 2. '+' and '-' are treated the same.
-        /// 3. '/' and '_' are treated the same.
-        /// </remarks>
-        internal static void Decode(ReadOnlySpan<char> input, int offset, int length, byte[] output)
```

Comment:
I think this should also be added back, unless it was verified that it it isn't used in other SDKs?

Reply:
This is `internal` so it should be fine, right?

Reply:
Internals can still be used between the projects here or from our other SDKs, so we treat them as public.